### ISSUE TITLE
Provisioning multiple dashboard folders results in "database is locked" error

### ISSUE THREAD

**Issue by plodwick** [grafana/grafana#16395]
Using the following yaml, on boot we get the following error message:

_t=2019-04-04T16:43:17+0000 lvl=eror msg="failed to search for dashboards" logger=provisioning.dashboard type=file name=beta error="database is locked"_

**The dashboard is NOT provisioned until the next updateIntervalSeconds pass.**

**YAML:**
apiVersion: 1

providers:
- name: 'released'
  orgId: 1
  folder: 'Released'
  type: file
  disableDeletion: true
  updateIntervalSeconds: 3600 #how often Grafana will scan for changed dashboards
  options:
    path: /etc/grafana/provisioning/dashboards/released
- name: 'beta'
  orgId: 1
  folder: 'Beta'
  type: file
  disableDeletion: false
  updateIntervalSeconds: 3600 #how often Grafana will scan for changed dashboards
  options:
    path: /etc/grafana/provisioning/dashboards/beta

**What happened**: Dashboard is not provisioned for 1 hour.  (3600 seconds)

**What you expected to happen**: Dashboard provisioned right away.

**How to reproduce it (as minimally and precisely as possible)**:

- Use grafana/grafana docker image
- Add dashboards to beta and release directories (according to above YAML) (ADD dashboards /etc/grafana/provisioning/dashboards)
- Start docker container

**Environment**: docker container
- Grafana version: 6.1.0
- Data source type & version: N/A
- OS Grafana is installed on: N/A
- User OS & Browser: N/A
- Grafana plugins: N/A
- Others:

This feels like it might be the root cause for issue [13316](https://github.com/grafana/grafana/issues/13316).  They reply "new dashboards appear after a couple of clicks."  This could be due to a smaller updateIntervalSeconds configuration.

**Comment by marefr** [comment]
How many dashboards do you have in each released and beta folder? Are you running docker and using storage for Grafana container from network folder (NFS)?

**Comment by plodwick** [comment]
@marefr I just have one dashboard in each directory.   I found the problem with Docker running on OpenShift using NFS.  I reproduced it running on Docker for Windows.  This morning I was able to reproduce running directly on Windows.  I downloaded https://dl.grafana.com/oss/release/grafana-6.1.0.windows-amd64.zip, extracted, added this YAML file in the provisioning dashboard directory.   The first folder showed up immediately and I got the error.  The second folder appeared in 120 seconds.

I **WAS NOT** able to reproduce this running directly with version **5.4.3**, but did get this message which makes me suspect that the retry was removed for 6.1.0

[32mINFO[0m[04-05|09:05:38] Database table locked, sleeping then retrying [32mlogger[0m=sqlstore [32mretry[0m=1

**YAML**:
apiVersion: 1

providers:
- name: 'released'
  orgId: 1
  folder: 'Released'
  type: file
  disableDeletion: false
  updateIntervalSeconds: 120 #how often Grafana will scan for changed dashboards
  options:
    path: c:\temp\dashboards\released
- name: 'beta'
  orgId: 1
  folder: 'Beta'
  type: file
  disableDeletion: false
  updateIntervalSeconds: 120 #how often Grafana will scan for changed dashboards
  options:
    path: c:\temp\dashboards\beta
